malariagen_data/anopheles.py

```diff
@@ -1479,14 +1486,21 @@ def _snp_sites(
         chunks,
     ):
         assert isinstance(region, Region), type(region)
-        root = self.open_snp_sites()
-        z = root[f"{region.contig}/variants/{field}"]
-        ret = da_from_zarr(z, inline_array=inline_array, chunks=chunks)
+
+        ret = self._snp_sites_for_contig(
+            contig=region.contig, field=field, inline_array=inline_array, chunks=chunks
+        )
+
         if region.start or region.end:
             if field == "POS":
-                pos = z[:]
+                pos = ret
             else:
-                pos = root[f"{region.contig}/variants/POS"][:]
+                pos = self._snp_sites_for_contig(
+                    contig=region.contig,
+                    field="POS",
+                    inline_array=inline_array,
+                    chunks=chunks,
+                )
             loc_region = locate_region(region, pos)
```

Comment:
This looks good but just noting there is a subtle change here, because `pos` is now a dask array whereas previously it was a numpy array. This should be OK, because it will I believe get converted to a numpy array (i.e., computed) within the `locate_region` function (actually within `allel.SortedIndex`). We want this to happen, because performing any kind of binary search on a dask array is likely to have poor performance.

Reply:
Yeah I meant to highlight this in the PR and check with you if it was OK but forgot, perhaps it would be good for me to build the PR message as I'm doing the coding.

Reply:
No worries at all, just thought I'd flag in case we notice any performance issues.